platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md

```diff
@@ -0,0 +1,158 @@
+This flush operation has to complete before the associated round can be handled, so there is also a wire from the
+PCES writer to the `RoundDurabilityBuffer` to prevent handling of a round until all events within the round have been
+written to disk.
+
+We propose to remove these extra complications (PCES Sequencer, `RoundDurabilityBuffer`, consensus control of flush,
+"keystone" events). Through experimentation, we have determined that the `FileChannel` API is fast enough to make the
+writing of each event a synchronous operation. This means we can send events directly from event intake into the PCES,
+and directly from PCES into consensus and, for self-events, into the gossip engine.
+
+This allows us to remove the flush call and the PCES Sequencer, while also closing the loophole causing branching.
+
+#### Alternatives Considered
+
+##### Force the PCES to flush after the last self event is created when freezing
+
+If the PCES is forced to flush after the last self event is created when freezing, then the PCES will be in a consistent
+state when restarted. However, this does not solve the problem if a deterministic crash affects all nodes, and it does
+not get us closer to the end-state of a simpler system.
+
+##### Other Methods For Writing
+
+We also explored several other APIs for writing events to disk (for example, memory mapped files). Of all the APIs we
+explored, the `FileChannel` API was the fastest and simplest to use.
+
+---
+
+## Changes
+
+### Architecture and/or Components
+
+At a high level, the architecture remains the same: events are created by the "Event Creator System", they are gossiped
+by the "Gossip System", events are handled by the "Event Intake System", persisted by the
+"Pre-Consensus Recording System", and come to consensus in the "Consensus System".
+
+![consensusPlatformArch](consensusPlatformArch.png)
+
+In this case, we are only interested in _self events_. These are created by the `Event Creator System` and passed to
+the `Event Intake System`. Here, they are ordered and validated and sent to the `Preconsensus Recording System`. Here,
+they will be immediately persisted and then sent to the `Consensus System` and to the `Gossip System`.
+
+In the short term, we will not send any events to the gossip system before they pass through the PCES system. When we
+eventually deploy gossip algorithms that call for out of order transmission (e.g. chatter), we may decide to expose
+other events (i.e. events not created by ourselves) to gossip before they pass through the orphan buffer.
+
+### Impact on Wiring
+
+#### PCES Writer
+
+PCES Writer will be changed to use `FileChannel`. All writes will be synchronous. The PCES writer will accept an event
+as input, and will return that same event once it has been written to disk. The `flush` API will be removed
+(or, at least, no longer used by the consensus engine. It may be required for clean shutdown,
+which will be handled by another issue).
```

Comment:
Does this mean the PCES Writer will be a direct scheduler component?

Reply:
No, it will still be sequential. It will receive events as intake, write those events on its thread, and then pass events to its output after they are durable. What I mean by "synchronous" in this context is that the system will wait until the PCES writer has finished writing prior to using the events, as opposed to using them as they are written asynchronously.